docs/dev/README.md

```diff
@@ -0,0 +1,45 @@
+# Introduction
+
+The **Firefox Translations Web Extension** ties together the various components that together form the **Firefox Translations** feature.
+
+The source code is available at [https://github.com/mozilla-extensions/firefox-translations/]().
+
+## About this guide
+
+This guide is meant for developers of the **Firefox Translations Web Extension**.
+
+For developer documentation related to any other part than the Web Extension (and what those parts may be), check the section called `Related repositories` below.
+
+For user documentation on the **Firefox Translations** feature, refer to [Firefox Translations User Guide](../user/index.html).
```

Comment:
The link `../user/index.html` is not valid. I believe you meant  `../user/usage.md`?

Reply:
It is valid in the generated html docs, see https://motin.github.io/firefox-translations/dev/index.html

Reply:
If I click on `Firefox Translations User Guide` link in [about-this-guide](https://github.com/motin/firefox-translations/tree/improve-docs-and-clean-up/docs/dev#about-this-guide), it shows 404 error.

I am not sure how html doc is picking up the right page but github README isn't.

Reply:
Click on that link in https://motin.github.io/firefox-translations/dev/index.html and see if it works for you. This is the same setup that Glean uses, see https://mozilla.github.io/glean/book/index.html

Reply:
I want to emphasize that this works in html pages but not on github pages.

If this documentation is intended to be consumed only via html pages then things are fine but if anyone is browsing the github repo and following the documentation then these links will be broken 😄

Now, I don't know whether this is acceptable or not. Is there a way that we can maintain it both ways?

Reply:
To clarify the term GitHub pages, I am using it to refer to [GitHub Pages](https://pages.github.com/) and the documentation is only intended to be read via the rendered version, just like Glean's documentation is. To clarify this, we can add a html/markdown comment on all source-pages with a link to the rendered version of the documentation, in case someone browses the repo and looking through the documentation.